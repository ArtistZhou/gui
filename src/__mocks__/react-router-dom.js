import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
