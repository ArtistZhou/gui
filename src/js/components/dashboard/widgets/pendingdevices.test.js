import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { createMount } from '@material-ui/core/test-utils';
import PendingDevices from './pendingdevices';

describe('PendingDevices Component', () => {
  it('renders correctly', () => {
    const tree = createMount()(
      <MemoryRouter>
        <PendingDevices />
      </MemoryRouter>
    );
    expect(tree.html()).toMatchSnapshot();
  });
});
