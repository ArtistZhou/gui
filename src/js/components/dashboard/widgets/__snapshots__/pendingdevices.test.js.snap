// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <a
    className="MuiButtonBase-root-11 MuiFab-root-1 MuiFab-primary-3"
    href="/devices/pending"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    style={
      Object {
        "left": "15px",
        "position": "absolute",
        "top": "-28px",
        "zIndex": "1",
      }
    }
    tabIndex="0"
  >
    <span
      className="MuiFab-label-2"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-14"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root-23"
    />
  </a>
  <div
    className="notActive widget "
    onClick={[Function]}
  >
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <div
        className="widgetMainContent align-right"
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <div
          className="header"
        >
          Pending devices
        </div>
        <div
          className="counter"
        />
      </div>
      <span
        className="link"
      >
        View details
      </span>
    </div>
  </div>
</div>
`;
